test/pages/HasConfidentialInformationPageSpec.scala

```diff
@@ -28,4 +31,18 @@ class HasConfidentialInformationPageSpec extends PageBehaviours {
 
     beRemovable[Boolean](HasConfidentialInformationPage)
   }
+  "cleanup" - {
+    "should reset ConfidentialInformationPage" - {
+      "when HasConfidentialInformationPage is changed to No" in {
+        val emptyUserAnswers = UserAnswers("id", DraftId(1))
+
+        val ua = emptyUserAnswers
+          .set(ConfidentialInformationPage, "secret")
+          .get
+
+        HasConfidentialInformationPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
```

Comment:
We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).